src/editors/_baseEditor.js

```diff
@@ -12,258 +14,340 @@ export const EditorState = {
-
-  this.init();
-}
+class BaseEditor {
+  constructor(instance) {
+    /**
+     * A reference to the source instance of the Handsontable.
+     */
+    this.hot = instance;
+    /**
+     * A reference to the source instance of the Handsontable.
+     * @deprecated
+     */
+    this.instance = instance;
+    /**
+     * Editor's state.
+     */
+    this.state = EditorState.VIRGIN;
 
-BaseEditor.prototype._fireCallbacks = function(result) {
-  if (this._closeCallback) {
-    this._closeCallback(result);
+    /**
+     * Flag to store information about editor's opening status.
+     * @private
+     */
+    this._opened = false;
+    /**
+     * Defines the editor's editing mode. When false, then an editor works in fast editing mode.
+     * @private
+     */
+    this._fullEditMode = false;
+    /**
+     * Callback to call after closing editor.
+     */
     this._closeCallback = null;
+
+    this.init();
   }
-};
 
-BaseEditor.prototype.init = function() {};
+  /**
+   * Fires callback after closing editor.
+   *
+   * @private
+   * @param {Boolean} result
+   */
+  _fireCallbacks(result) {
+    if (this._closeCallback) {
+      this._closeCallback(result);
+      this._closeCallback = null;
+    }
+  }
 
-BaseEditor.prototype.getValue = function() {
-  throw Error('Editor getValue() method unimplemented');
-};
+  /**
+   * Initializes an editor's intance.
+   */
+  init() {}
 
-BaseEditor.prototype.setValue = function() {
-  throw Error('Editor setValue() method unimplemented');
-};
+  /**
+   * Required method to get current value from editable element.
+   */
+  getValue() {
+    throw Error('Editor getValue() method unimplemented');
+  }
 
-BaseEditor.prototype.open = function() {
-  throw Error('Editor open() method unimplemented');
-};
+  /**
+   * Required method to set new value into editable element.
+   */
+  setValue() {
+    throw Error('Editor setValue() method unimplemented');
+  }
 
-BaseEditor.prototype.close = function() {
-  throw Error('Editor close() method unimplemented');
-};
+  /**
+   * Required method to open editor.
+   */
+  open() {
+    throw Error('Editor open() method unimplemented');
+  }
 
-BaseEditor.prototype.prepare = function(row, col, prop, td, originalValue, cellProperties) {
-  this.TD = td;
-  this.row = row;
-  this.col = col;
-  this.prop = prop;
-  this.originalValue = originalValue;
-  this.cellProperties = cellProperties;
-  this.state = EditorState.VIRGIN;
-};
+  /**
+   * Required method to close editor.
+   */
+  close() {
+    throw Error('Editor close() method unimplemented');
+  }
 
-BaseEditor.prototype.extend = function() {
-  const baseClass = this.constructor;
+  /**
+   * Prepares editor's meta data.
+   *
+   * @param {Number} row
+   * @param {Number} col
+   * @param {Number|String} prop
+   * @param {HTMLTableCellElement} td
+   * @param {*} originalValue
+   * @param {Object} cellProperties
+   */
+  prepare(row, col, prop, td, originalValue, cellProperties) {
+    this.TD = td;
```

Comment:
For better readability and performance it's better to define all properties used in that class in the constructor (to notify JS engines so that they can better optimize internal data structures).